### ISSUE TITLE
Null safety support

### ISSUE THREAD

**Issue by stepanzalis** [aloisdeniel/flutter_sheet_localization#38]
Hello 👋 

first of all, thank you for the package, it helped me a lot with sharing translation with others. Do you consider to maintain it and migrate it to null-safety? 

Thank you 🙏

**Comment by stepanzalis** [comment]
Oh, I found the related issue already. #33 
Sorry

**Comment by Dokotela** [comment]
I've tried to migrate this to null safety, and opened a PR. But if anyone wanted to test it and let me know, that would be helpful. https://github.com/Dokotela/flutter_sheet_localization

**Comment by aloisdeniel** [comment]
In the meanwhile, I published a 3.0.0 version yesterday with null-safety (for the main package) support and upgraded dependencies.

Let me know after upgrading if it works for you!

**Comment by 4mitabh** [comment]
@aloisdeniel the 3.0.0 version is supposed to produce null safe localizations.g.dart, right?

It seems to generate the same code for me- the same as 2.x versions.
Perhaps I'm missing something...

**Comment by stepanzalis** [comment]
@aloisdeniel Thanks. I'll check it out.

**Comment by rtoshmukhamedov** [comment]
@aloisdeniel Can you upgrade to the latest source_gen: ^1.0.0 and dart_style: ^2.0.0. Because other packages are requiring them.

**Comment by stepanzalis** [comment]
@aloisdeniel Can you please update source_gen: ^1.0.0 and dart_style: ^2.0.0? Seems that generating does not change anything and got some errors.

**Comment by stepanzalis** [comment]
Generator produces the same code as version 2.x as mentioned @4mitabh. Build crashes when you use Dart 2.12.

**Comment by aloisdeniel** [comment]
Sorry for the delay, I was on a big refactoring pass.

A new version is available. Please tell me if it works better on your side!

Note that there are a few changes to do to your delegate class implementation (see the README for more details).

**Comment by mboe** [comment]
I tried the new version (and made the changes in `localization.dart` file) but `AppLocalizations` is not a valid class anymore.

**Comment by aloisdeniel** [comment]
Ah yes, I forgot this part of the documentation.

The new recommended way of accessing localizations is :

```dart
extension AppLocalizationsExtensions on BuildContext {
  AppLocalizationsData get localizations {
    return Localizations.of<AppLocalizationsData>(this, AppLocalizationsData)!;
  }
}
```

If you want to keep support in your code base you can define a class :

```dart
abstract class AppLocalizations {
  static AppLocalizationsData of(BuildContext context) {
    return Localizations.of<AppLocalizationsData>(context, AppLocalizationsData)!;
  }
}
```

**Comment by mboe** [comment]
Yes, thank you for the clarification but that not sufficient. There is no `.languages` property (on both classes) but according to your documentation it should (see Part 4 'Configure your App') and I do miss the `getByKey` function. I used it a lot because the actual key was determined during runtime.

**Comment by aloisdeniel** [comment]
Il do a full pass on documentation by the end of the week to make sure everything is up to date.

In the meantime, you can refer to [the provided example](https://github.com/aloisdeniel/flutter_sheet_localization/tree/master/flutter_sheet_localization_generator/example/lib).

**Comment by mboe** [comment]
Hi. Thanks a lot. I was able to make my code run again. The only issue remaining is the absence of the `getByKey` function (or anything similar). I use this in a generic class where the actual field keys were derived from database responses. This was very dynamic and slim and I'd like to have a similar functionality again. Can you please check?

**Comment by mboe** [comment]
I'm not sure whether my short explanation was enough for you. Imaging you have a widget class (e.g. a table) which shall visualize data fetched from a (lets say) database. The data shall have localized header fields. The widget class does not know anything about the columns and therefore the header fields. The caller code is dynamic, too and does not need to know all about the data. Therefore only the database data itself and the localization sheet should be in sync and without any code change more or less columns can be displayed. This worked perfectly with the @getByKey @ function. Without I have to "know" the possible columns in the code. Every change of the data must lead to a code change, too. That's not so nice.
